src/test/java/bio/terra/common/iam/SamUserAuthenticatedRequestFactoryTest.java

```diff
@@ -0,0 +1,74 @@
+package bio.terra.common.iam;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import bio.terra.common.exception.UnauthorizedException;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
+import org.broadinstitute.dsde.workbench.client.sam.auth.OAuth;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+
+@Tag("unit")
+public class SamUserAuthenticatedRequestFactoryTest {
+
+  private static final String EMAIL_ADDRESS = "test@example.com";
+  private static final String SUBJECT_ID = "Subject";
+  private static final TokenAuthenticatedRequest TOKEN =
+      new TokenAuthenticatedRequest.Builder().setToken("0123.456-789AbCd").build();
+
+  @Test
+  public void enabledUser() throws ApiException {
+    TestFactory factory = new TestFactory();
+    when(factory.usersApi.getUserStatusInfo())
+        .thenReturn(
+            new UserStatusInfo().userEmail(EMAIL_ADDRESS).userSubjectId(SUBJECT_ID).enabled(true));
+
+    SamUserAuthenticatedRequest outReq = factory.from(TOKEN, new ApiClient());
+    assertEquals(EMAIL_ADDRESS, outReq.getEmail());
+    assertEquals(SUBJECT_ID, outReq.getSubjectId());
+    assertEquals(TOKEN, outReq.getTokenRequest());
+  }
+
+  @Test
+  public void disabledUser() throws ApiException {
+    TestFactory factory = new TestFactory();
+    when(factory.usersApi.getUserStatusInfo())
+        .thenReturn(
+            new UserStatusInfo().userEmail(EMAIL_ADDRESS).userSubjectId(SUBJECT_ID).enabled(false));
+
+    assertThrows(UnauthorizedException.class, () -> factory.from(TOKEN, new ApiClient()));
+  }
+
+  @Test
+  public void notFoundUser() throws ApiException {
+    TestFactory factory = new TestFactory();
+    when(factory.usersApi.getUserStatusInfo())
+        .thenThrow(new ApiException(HttpStatus.NOT_FOUND.value(), "not found"));
+
+    assertThrows(UnauthorizedException.class, () -> factory.from(TOKEN, new ApiClient()));
+  }
+
+  private static class TestFactory extends SamAuthenticatedUserRequestFactory {
```

Comment:
Subclassing is fine but I think using `spy()` to do this is a more idiomatic way to use the mocks library. For example: https://github.com/DataBiosphere/terra-data-catalog/blob/main/service/src/test/java/bio/terra/catalog/iam/SamServiceTest.java